wlib/LangRef.cpp

```diff
@@ -711,18 +727,147 @@ class CLangRef : public ILangRef, public CUnknown
+    CString & GetLangNames(int cat, CString &result)
+    {
+        result = GetLangName(cat);
+        return result;
+    }
+    virtual void init()
+    {
+        appendColor(CLanguageColor(SCE_H_DEFAULT, _T("default")));
+        m_defaultColor = &m_color[SCE_H_DEFAULT];
+
+        const COLORREF red = RGB(0xFF, 0, 0);
+        const COLORREF offWhite = RGB(0xFF, 0xFB, 0xF0);
+        const COLORREF darkGreen = RGB(0, 0x80, 0);
+        const COLORREF darkBlue = RGB(0, 0, 0x80);
+
+        appendColor(CLanguageColor(SCE_H_TAG, _T(""), darkBlue));
+        appendColor(CLanguageColor(SCE_H_TAGUNKNOWN, _T(""), red));
+        appendColor(CLanguageColor(SCE_H_ATTRIBUTE, _T(""), darkBlue));
+        appendColor(CLanguageColor(SCE_H_ATTRIBUTEUNKNOWN, _T(""), red));
+        appendColor(CLanguageColor(SCE_H_NUMBER, _T(""), RGB(0x80, 0, 0x80)));
+        appendColor(CLanguageColor(SCE_H_DOUBLESTRING, _T(""), RGB(0, 0x80, 0)));
+        appendColor(CLanguageColor(SCE_H_SINGLESTRING, _T(""), RGB(0, 0x80, 0)));
+        appendColor(CLanguageColor(SCE_H_OTHER, _T(""), RGB(0x80, 0, 0x80)));
+        appendColor(CLanguageColor(SCE_H_COMMENT, _T(""), RGB(0x80, 0x80, 0)));
+        appendColor(CLanguageColor(SCE_H_ENTITY, _T(""), RGB(0x80, 0, 0x80)));
+
+        appendColor(CLanguageColor(SCE_H_TAGEND, _T(""), darkBlue));
+        appendColor(CLanguageColor(SCE_H_XMLSTART, _T(""), darkBlue));
+        appendColor(CLanguageColor(SCE_H_XMLEND, _T(""), darkBlue));
+        appendColor(CLanguageColor(SCE_H_SCRIPT, _T(""), darkBlue));
+        appendColor(CLanguageColor(SCE_H_ASP, _T(""), RGB(0x4F, 0x4F, 0), RGB(0xFF, 0xFF, 0)));
+        appendColor(CLanguageColor(SCE_H_ASPAT, _T(""), RGB(0x4F, 0x4F, 0), RGB(0xFF, 0xFF, 0)));
+    }
+    void appendColor(const CLanguageColor & color) {
+        m_color[color.GetCategoryID()] = color;
+        m_colorRowToCategoryID.push_back(color.GetCategoryID());
+    }
+};
+
+class CJSLangRef : public CLangRef
+{
+    int GetLexerType()
+    {
+        return SCLEX_CPP;
+    }
+};
+
 boost::recursive_mutex g_langRef_mutex;
-StlLinked<CLangRef> g_langRef;
+std::map<std::_tstring, StlLinked<CLangRef> > g_langRef;
 
-ILangRef * CreateEclLangRef()
+ILangRef * CreateLangRef(IAttributeType * type)
 {
 	boost::recursive_mutex::scoped_lock proc(g_langRef_mutex);
-	if (!g_langRef)
-	{
-		g_langRef = new CLangRef;
-		g_langRef->init();
-	}
-	return g_langRef;
+	if (!g_langRef[type->GetRepositoryCode()])
+	{
+        if (boost::algorithm::equals(type->GetRepositoryCode(), _T("html"))) {
+            g_langRef[type->GetRepositoryCode()] = new CHTMLLangRef();
+        }
+        else if (boost::algorithm::equals(type->GetRepositoryCode(), _T("js"))) {
+            g_langRef[type->GetRepositoryCode()] = new CJSLangRef();
+        }
+        else {
+            g_langRef[type->GetRepositoryCode()] = new CECLLangRef();
+        }
+        g_langRef[type->GetRepositoryCode()]->init();
+	}
+	return g_langRef[type->GetRepositoryCode()];
 };
```

Comment:
Add new clases for each new known language (this is probably _not_ the way we will do it - I suspect we will instead have a new lang.xml file per language.